### PR TITLE
feat: new pagination design

### PR DESCRIPTION
**Pull request by patricio0312rev**
# [[general] implement new pagination design](https://app.clickup.com/t/85ztypggv)

## Summary

- New pagination design has been implemented to match our figma files.

## Views

- 320px:

image

- 640px:

image

- 768px:

image

- 960px:

image

- 1024px:

image

- 1280px:

image

- 1440px and more:

image

## Checklist

-   [ ] I checked my UI changes against the design and there are no notable differences
-   [ ] I checked my UI changes for any responsiveness issues
-   [ ] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
-   [ ] I added a short description on how to test this PR _(if necessary)_
-   [ ] I added a storybook entry for the component that was added _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

**Review by ItsANameToo — CHANGES_REQUESTED** [review]
some issues UI wise:
- pagination icons should be 12x12, they are currently 16x16
- the page limit dropdown is very small, it should be as wide as the entry itself
impl:
image

design:
image
- search close button on desktop gets squashed. Also the search input should be larger and show placeholder by default instead of the page
image
- when opening the search on desktop, the entire pagination row moves down slightly
- on mobile the prev/next etc buttons are inconsistent when they appear and not in line with design. They currently hide the first/prev buttons on page 1, but on the last page it should hide both next/last buttons and it only hides the "last" button
- between 375 and above the pagination does not always fit
image
- on 375 and below, the search should not open where the page number is but below it instead (to replace the navigation items)
design:
image

**Review by samharperpittam — CHANGES_REQUESTED** [review]
At breakpoint 640, the pagination should be the mobile variant.

Design
image


Implementation
image

---

On desktop when clicking the 3 dots to enter a page number, the search field is incredibly small.

Design
image


Implementation
image

**Review by samharperpittam — CHANGES_REQUESTED** [review]
At 640, is there a reason why the numeric dropdown doesn't display the text `Items`?
image

**Review by samharperpittam — CHANGES_REQUESTED** [review]
Also, is it possible to modify this so the mobile view starts at 375 down instead of 320 please?

image

Edit: can ignore this comment.

**Review by samharperpittam — APPROVED** [review]

**Review by samharperpittam — APPROVED** [review]

File: resources/js/Components/Pagination/SelectPageLimit.tsx [hunk 1]
```diff
@@ -8,17 +8,19 @@ export const SelectPageLimit = ({
...
     onChange?: ({ value, url }: { value: string | number; url: string }) => void;
+    suffix: string;
 }): JSX.Element => {
     const { t } = useTranslation();
 
     return (
         <Listbox
-            className="w-full sm:w-40 md:w-[13.5rem]"
-            optionsClassName="right-0 w-full sm :w-40 bottom-14"
+            className="w-full xs:w-[12rem] sm:flex sm:justify-between"
```
Comment (ItsANameToo) [inline resources/js/Components/Pagination/SelectPageLimit.tsx:22]:
```suggestion
            className="w-full xs:w-48 sm:flex sm:justify-between"
```

File: resources/js/Components/Pagination/PreviousPageLink.tsx [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import { ButtonLink } from "@/Components/Buttons/ButtonLink";
 
-export const PreviousPageLink = ({ href }: { href: string }): JSX.Element => (
+export const PreviousPageLink = ({ href }: { href: string | null }): JSX.Element => (
     <ButtonLink
-        href={href}
+        disabled={href == null}
```
Comment (ItsANameToo) [inline resources/js/Components/Pagination/PreviousPageLink.tsx:5]:
```suggestion
        disabled={href === null}
```

File: resources/js/Components/Pagination/Pagination.tsx [hunk 3]
```diff
@@ -67,15 +68,28 @@ export const Pagination = <T,>({ data, ...properties }: PaginationProperties<T>)
...
+    const handleClick = (): void => {
+        setShowInput(true);
+        setPage(data.meta.current_page.toString());
+    };
+
     return (
         <nav
             aria-label="Pagination"
             data-testid="Pagination"
+            className="!m-0 w-full sm:w-fit"
```
Comment (ItsANameToo) [inline resources/js/Components/Pagination/Pagination.tsx:80]: why did it need the `!` for the margin?

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**